### PR TITLE
Cancel user retirement on password reset story

### PR DESCRIPTION
**Pull request by devalih**
### Description

TASK: [ASANA LINK](https://app.asana.com/X/XXXXXX/XXXXXX)

Add a description of your changes here.

### Notes
- Example: This PR will not address x,y, and z, because of a, b, and c.

### Testing
- [ ] i18n
- [ ] RTL
- [ ] safecommit shows 0 violations
- [ ] Unit, integration, acceptance tests as appropriate
- [ ] Performance
- [ ] Database migrations are backwards-compatible

### Related PRs
- Example: [Configuration Pull #XX](github.com/edraak/configuration/pull/XX)

### CODE REVIEW

File: common/djangoapps/student/forms.py [hunk 1]
```diff
@@ -47,11 +48,22 @@ def clean_email(self):
...
+                raise forms.ValidationError(self.error_messages['unknown'])
+            else:
+                retirement_status.user.email = retirement_status.original_email
+                retirement_status.user.username = retirement_status.original_username
+                retirement_status.user.name = retirement_status.original_name
+
+                self.users_cache = [retirement_status.user]
+        # if any((user.password.startswith(UNUSABLE_PASSWORD_PREFIX))
+        #        for user in self.users_cache):
+        #     raise forms.ValidationError(self.error_messages['unusable'])
```
Comment (ghost) [inline common/djangoapps/student/forms.py:78]:
```suggestion
```

File: common/djangoapps/student/admin.py [hunk 2]
```diff
@@ -227,7 +227,17 @@ class UserProfileInline(admin.StackedInline):
 class UserAdmin(BaseUserAdmin):
     """ Admin interface for the User model. """
     inlines = (UserProfileInline,)
-
+    # EDRAAK SPCIFIC overrid to remove groups from the filters
```
Comment (ghost) [inline common/djangoapps/student/admin.py:230]:
```suggestion
    # EDRAAK SPECIFIC override to remove groups from the filters
```

**Review by ghost — APPROVED** [review]
I don't think this feature is GDPR compliant.

GDPR user deletion is supposed to [be permanent](https://gdpr-info.eu/art-17-gdpr/) and not to be cancelled.

> ... erasure of personal data concerning him or her without undue delay and the controller ...

However, if you think this is a must then let's go with it.

**Comment by devalih** [comment]
We only allow cancelling a retirement request, we don't recover a fully retired user, it's not possible the data is deleted permanently

**Pull request merged**